homeassistant/components/sensor/forecast.py

```diff
@@ -104,16 +106,21 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
             "Please check your settings for Forecast.io.")
         return False
 
+    units = config.get('units')
```

Comment:
You can specify the default value if key does not exist by giving a second argument to the `get` method:
```python
units = config.get('units', 'auto')
```

That being said, instead of defaulting to auto, I think we should have it default to `si` or `us` based on the temperature preference in Home Assistant.
```python
if 'units' in config:
  units = config['units']
elif hass.config.temperature_unit == TEMP_CELCIUS:
  units = 'si'
else:
  units = 'us'
```